internal/storage/store.go

```diff
@@ -0,0 +1,540 @@
+		item, ok := itemsValue.Index(i).Addr().Interface().(runtime.Object)
+		if !ok {
+			return nil, storage.NewInternalErrorf("unexpected item type: %T", itemsValue.Index(i).Addr().Interface())
+		}
+
+		events = append(events, watch.Event{
+			Type:   watch.Added,
+			Object: item,
+		})
+	}
+
+	return s.broadcaster.WatchWithPrefix(events)
+}
+
+// Get unmarshals object found at key into objPtr. On a not found error, will either
+// return a zero object of the requested type, or an error, depending on 'opts.ignoreNotFound'.
+// Treats empty responses and nil response nodes exactly like a not found error.
+// The returned contents may be delayed, but it is guaranteed that they will
+// match 'opts.ResourceVersion' according 'opts.ResourceVersionMatch'.
+func (s *store) Get(ctx context.Context, key string, opts storage.GetOptions, objPtr runtime.Object) error {
+	name, namespace := extractNameAndNamespace(key)
+	if name == "" || namespace == "" {
+		return storage.NewInternalErrorf("invalid key: %s", key)
+	}
+
+	if err := runtime.SetZeroValue(objPtr); err != nil {
+		return storage.NewInternalErrorf("unable to set objPtr zero value: %v", err)
+	}
+
+	query, args, err := sq.Select("*").
+		From(s.table).
+		Where(sq.Eq{"name": name, "namespace": namespace}).
+		ToSql()
+	if err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	objectRecord := &objectSchema{}
+	if err := s.db.GetContext(ctx, objectRecord, query, args...); err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			if opts.IgnoreNotFound {
+				return nil
+			}
+
+			return storage.NewKeyNotFoundError(key, 0)
+		}
+		return storage.NewInternalError(err.Error())
+	}
+
+	err = json.Unmarshal(objectRecord.Object, objPtr)
+	if err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	return nil
+}
+
+// GetList unmarshalls objects found at key into a *List api object (an object
+// that satisfies runtime.IsList definition).
+// The returned contents may be delayed, but it is guaranteed that they will
+// match 'opts.ResourceVersion' according 'opts.ResourceVersionMatch'.
+func (s *store) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error {
+	queryBuilder := sq.Select("*").From(s.table)
+	namespace := extractNamespace(key)
+	if namespace != "" {
+		queryBuilder = queryBuilder.Where(sq.Eq{"namespace": namespace})
+	}
+	query, args, err := queryBuilder.ToSql()
+	if err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	var objectRecords []objectSchema
+	if err := s.db.SelectContext(ctx, &objectRecords, query, args...); err != nil {
+		return storage.NewInternalError(err.Error())
+	}
+
+	itemsValue, err := getItems(listObj)
+	if err != nil {
+		return err
+	}
+
+	for _, objectRecord := range objectRecords {
+		obj := s.newFunc()
+		if err := json.Unmarshal(objectRecord.Object, obj); err != nil {
+			return storage.NewInternalError(err.Error())
+		}
+
+		ok, err := opts.Predicate.Matches(obj)
```

Comment:
Should we plan to use SQL selectors in the future to filter the results at the db level instead of using this approach? We don't have to fix it right now, just to track this "technical debt"

Reply:
I take it back, it's already tracked